repose-aggregator/services/atom-feed/impl/src/main/scala/org/openrepose/nodeservice/atomfeed/impl/AtomFeedServiceImpl.scala

```diff
@@ -0,0 +1,230 @@
+import org.openrepose.docs.repose.atom_feed_service.v1.{AtomFeedServiceConfigType, OpenStackIdentityV2AuthenticationType}
+import org.openrepose.nodeservice.atomfeed.impl.actors.NotifierManager._
+import org.openrepose.nodeservice.atomfeed.impl.actors._
+import org.openrepose.nodeservice.atomfeed.{AtomFeedListener, AtomFeedService, AuthenticatedRequestFactory}
+import org.springframework.beans.factory.annotation.Value
+
+import scala.collection.JavaConversions._
+import scala.language.postfixOps
+
+@Named
+class AtomFeedServiceImpl @Inject()(@Value(ReposeSpringProperties.CORE.REPOSE_VERSION) reposeVersion: String,
+                                    @Value(ReposeSpringProperties.NODE.CLUSTER_ID) clusterId: String,
+                                    @Value(ReposeSpringProperties.NODE.NODE_ID) nodeId: String,
+                                    configurationService: ConfigurationService)
+  extends AtomFeedService with LazyLogging {
+
+  import AtomFeedServiceImpl._
+
+  private val actorSystem: ActorSystem = ActorSystem.create("AtomFeedServiceSystem")
+
+  private var isServiceEnabled = false
+  private var feedActors: Map[String, FeedActorPair] = Map.empty
+  private var listenerNotifierManagers: Map[String, ActorRef] = Map.empty
+  // note: feed readers creation/destruction determined by configuration updates
+  // note: notifier managers creation/destruction determined by registering/unregistering
+
+  @PostConstruct
+  def init(): Unit = {
+    logger.info("Initializing and registering configuration listeners")
+    val xsdURL = getClass.getResource("/META-INF/schema/config/atom-feed-service.xsd")
+
+    configurationService.subscribeTo(
+      DEFAULT_CONFIG,
+      xsdURL,
+      AtomFeedServiceConfigurationListener,
+      classOf[AtomFeedServiceConfigType]
+    )
+    configurationService.subscribeTo(
+      SYSTEM_MODEL_CONFIG,
+      SystemModelConfigurationListener,
+      classOf[SystemModel]
+    )
+  }
+
+  @PreDestroy
+  def destroy(): Unit = {
+    logger.info("Unregistering configuration listeners and shutting down service")
+    configurationService.unsubscribeFrom(DEFAULT_CONFIG, AtomFeedServiceConfigurationListener)
+    configurationService.unsubscribeFrom(SYSTEM_MODEL_CONFIG, SystemModelConfigurationListener)
+
+    actorSystem.shutdown()
+  }
+
+  override def registerListener(feedId: String, listener: AtomFeedListener): String = {
+    val listenerId = java.util.UUID.randomUUID().toString
+    logger.info("Attempting to register a listener with id: " + listenerId + " to feed with id: " + feedId)
+
+    val notifierManager = feedActors.get(feedId) match {
+      case Some(FeedActorPair(manager, _)) =>
+        manager
+      case None =>
+        val manager = actorSystem.actorOf(Props[NotifierManager], name = feedId + NOTIFIER_MANAGER_TAG)
+        feedActors = feedActors + (feedId -> FeedActorPair(manager, None))
+
+        if (isServiceEnabled) {
+          manager ! ServiceEnabled
```

Comment:
Nope, I was just trying to mark that there was left to do for this comment stream.